openpgp-dsm/src/lib.rs

```diff
@@ -942,10 +954,12 @@ pub fn generate_key(
 pub struct DsmKeyInfo {
     name: String,
     kid: Uuid,
+    group_id: Uuid,
     object_type: ObjectType,
     created_at: SdkmsTime,
     last_used_at: SdkmsTime,
     fingerprint: String,
+    custom_metadata: HashMap<String, String>,
```

Comment:
this becomes redundant; `custom_metadata` contains `fingerprint`. If you implement **Option 2** of my above comment, here you could rename this field as `user_meta` and to populate it, extract `user_meta` from received custom metadata